### ISSUE TITLE
No-Slot Clock running and power-cycle can causes lock-up on power-cycle

### ISSUE THREAD

**Issue by tomcw** [AppleWin/AppleWin#751]
Repro:
- Boot NoSlotClockTest.dsk
- `-NS.CLOCK.UTILS`
- F2

It seems to be 50/50 whether it'll lock up.

```
FFFF: nop
0000: nop; nop; BRK  ; FFFE: FA C3
C3FA: (ROM code)
...
FA56: JMP (BRKV)  ; JMP ($3F0), 3F0: FF FF
```

A subsequent F2 will get it to boot.

Perhaps an MMU state isn't being reset correct?

**Comment by tomcw** [comment]
Hi @sicklittlemonkey / Nick - this is semi-interesting...

I was able to create a trace from the debugger of this going wrong (eg. enter debugger, F2 to power-cycle the Apple II machine, then from PC=FA62, "tf trace.txt"), to give this short trace of ~100 instructions:

[trace.txt](https://github.com/AppleWin/AppleWin/files/4748145/trace.txt)

Note at $C855, the instruction stream suddenly switches to $C3FA, which has the label `IRQ`.

So from the Visual Studio debugger, I set breakpoint on the 6502 IRQ handler, expecting an emulation bug to be erroneously generating an interrupt... but the VS breakpoint never fires.

Instead, at $C855, (again from VS) I step the 6502 emulation code:
- $C855 is in $Cxxx space, so the opcode is fetched by calling `IO_Cxxx(...)`, which is correct
- But half-way through this function is a call to `IsPotentialNoSlotClockAccess()`, which returns true
  - Correct, as INTCXROM is true && `AddrHi == 0xC8`
- This in turn calls `g_NoSlotClock.Read(int address, ...)` which returns true!
  -  address & 4 == true (ie. 0xC855 & 4 == 4)
  - `ClockRead(...)` return true, as `m_bClockRegisterEnabled == true`
- This returns 0x00, which a BRK, which is why the instruction stream suddenly vectors to IRQ! NB. it doesn't always return 0x00, sometimes it'll return 0x01 / ORA (zp,x).

Basically m_bClockRegisterEnabled isn't being reset when F2 (power-cycle) is pressed - easy to fix.

**Comment by tomcw** [comment]
An additional question is what happens on RESET?
Since the NSC chip just sits under a ROM chip, then it's not affected by the RESET signal.

**Comment by tomcw** [comment]
Ah, I see you add a comment for this in AppleWin / CNoSlotClock.cpp:
```
// SmartWatch reset - whether tied to system reset is component specific
```

**Comment by sicklittlemonkey** [comment]
Sorry for not responding to the original issue. I read it while in Japan on
"holiday" and then totally forgot about it!

I've never owned a NSC before, so I don't know if it was connected to reset
in the Apple II, but your fix makes sense.

Cheers,
Nick.

**Comment by tomcw** [comment]
No problem. btw, I also extended the save-state to include the NSC state and added a new `-no-nsc` command line switch. (c9ef6ecee1449824ef31fefc71e8e7c9afc88799)
